lib/govuk_navigation_helpers/grouped_related_links.rb

```diff
@@ -0,0 +1,48 @@
+module GovukNavigationHelpers
+  # Take a content item and group the related links according to an algorithm
+  # that is intended to display the related links into three groups, depending
+  # on how much they have in common with the main content item.
+  #
+  # @private
+  class GroupedRelatedLinks
+    attr_reader :content_item,
+
+      # This will contain related items that have the same `parent` (breadcrumb)
+      # as the content item.
+      :related_with_parent_in_common,
+
+      # This will contain related items that have a grandparent in common with
+      # the content item.
+      :related_with_grandparent_in_common,
+
+      # This will contain the related items that have a completely different
+      # parent/breadcrumb from the content item.
+      :related_with_nothing_in_common
```

Comment:
This name is a bit confusing to me. If they have nothing in common, how are they related? Would it make sense to rename this field?